lib/matplotlib/legend.py

```diff
@@ -472,6 +472,14 @@ def val_or_rc(val, rc_name):
                 loc = 'upper right'
         if isinstance(loc, str):
             loc = _api.check_getitem(self.codes, loc=loc)
+        elif isinstance(loc, tuple):
+            if len(loc) != 2:
+                raise ValueError(
+                    "loc must be string or pair of floats")
+            for loc_elt in loc:
+                # int can be converted to float, so ints are fine
+                if not isinstance(loc_elt, (float, int)):
+                    raise ValueError(f"{loc_elt} is not a valid value for loc")
```

Comment:
```suggestion
            if len(loc) != 2 or not all(isinstance(e, Number) for e in loc):
                raise ValueError("loc must be string or pair of numbers")
        else:        
            raise ValueError("loc must be string or pair of numbers")
```
- simplify to check for [Number](https://docs.python.org/3/library/numbers.html#numbers.Number)
- merge the len and the number checks
- error out for other input types

Reply:
I think we want `numbers.Real`, as in:
https://github.com/matplotlib/matplotlib/blob/cc85fb66da61f85259d7d99f06faa6285f4ed225/lib/matplotlib/axes/_secondary_axes.py#L95-L96

Reply:
Just incorporated these changes in the original commit. Thanks!